### DISCUSSION TITLE
next js router not working as expected

### DISCUSSION THREAD

**Discussion by Baryonae** [vercel/next.js#51128] [App Router] [answered]
whenever i use the <Link> component and put something in href, it always goes to the 404 page, i've searched many articles but none of them solved it out
for example, 
`<span className = 'px-2'><Link href = '/Auth'>Sign In</Link></span>`

there is a file in the same directory as 'Auth.js'
but clicking on the text doesnt get me there
will be really grateful if someone could figure this out

**Comment by devjiwonchoi** [comment, marked as answer]
In `App Router`, it is a convention to locate your route by folders, and name the file as `page.js | jsx | tsx`.

> Folders are used to define routes. A route is a single path of nested folders, following the file-system hierarchy from the root folder down to a final leaf folder that includes a page.js file. See [Defining Routes](https://nextjs.org/docs/app/building-your-application/routing/defining-routes).

Name a folder inside your `/app` segment as `Auth`, and name your page file inside as `page.js`.

Your file should look as follows:

```js
// /app/Auth/page.js

export default function Auth() {
  return <h1>This is Auth</h1>
}
```
![Screenshot 2023-06-11 at 10 49 51 PM](https://github.com/vercel/next.js/assets/120007119/d873447c-c069-4340-8c59-805143efe8cf)

## Reference
- [Roles of folders and files](https://nextjs.org/docs/app/building-your-application/routing#roles-of-folders-and-files)

**Reply by Baryonae** [reply]
Dude really thank you so much, it worked!
